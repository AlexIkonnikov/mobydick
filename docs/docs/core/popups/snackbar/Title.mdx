import Snackbar from '../assets/Snackbar.png';

<img src={Snackbar} width="50%"/>

Reusable Title component for Snackbar

## Example

```tsx
import {
  Button,
  createStyles,
  IPosition,
  SnackbarBase,
  Typography,
  usePopups,
  useStyles,
  View,
} from '@lad-tech/mobydick-core';

export const SnackbarWidget = () => {
  const [styles] = useStyles(styleFn);
  const {openPopup} = usePopups();
  return (
    <View style={styles.container}>
      <Typography font={'Regular-Primary-H5'}>Snackbar</Typography>
      <Button
        text={'Snackbar top'}
        onPress={() =>
          openPopup({
            Content: props => (
              <SnackbarBase {...props} position={IPosition.top}>
                <SnackbarBase.Title title={'Snackbar top title'} />
              </SnackbarBase>
            ),
          })
        }
      />
      <Button
        text={'Snackbar bottom'}
        onPress={() =>
          openPopup({
            Content: props => (
              <SnackbarBase {...props} position={IPosition.bottom}>
                <SnackbarBase.Title title={'Snackbar top bottom'} />
              </SnackbarBase>
            ),
          })
        }
      />
    </View>
  );
};

const styleFn = createStyles(({spaces}) => ({container: {gap: spaces.Space8}}));

```

## `Props`

### <div class="label required basic">Required</div>**`title`**

| TYPE   |
|:-------|
| string |

Title for Snackbar

### `titleStyles`

| TYPE                                                       |
|:-----------------------------------------------------------|
| [TextStyle](https://reactnative.dev/docs/text-style-props) |

Custom styles for container

### `titleFont`

| Type                                                                                                       | DEFAULT                |
|:-----------------------------------------------------------------------------------------------------------|:-----------------------|
| [TypographyProp](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/typography/types.ts#L49) | 'SemiBold-Secondary-M' |

Font for title
