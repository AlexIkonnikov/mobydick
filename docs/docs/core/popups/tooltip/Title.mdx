import Tooltip from '../assets/Tooltip.png';

<img src={Tooltip} width="50%"/>

Reusable Title component for Tooltip
## Example

```tsx
import {useRef} from 'react';

import {
  Button,
  createStyles,
  IPlacement,
  IPosition,
  TooltipBase,
  Typography,
  usePopups,
  useStyles,
  View,
} from '@lad-tech/mobydick-core';

export const TooltipWidget = () => {
  const ref = useRef(null);
  const [styles] = useStyles(styleFn);
  const {openPopup} = usePopups();

  return (
    <View style={styles.container}>
      <Typography font={'Regular-Primary-H5'}>Tooltip</Typography>
      <View ref={ref} style={styles.ref}>
        <Typography>Ref component</Typography>
      </View>
      <Button
        text={'Tooltip'}
        onPress={() =>
          openPopup({
            Content: props => (
              <TooltipBase
                position={IPosition.bottom}
                placement={IPlacement.center}
                refCurrent={ref}
                {...props}>
                <TooltipBase.Title title={'Tooltip title'} />
              </TooltipBase>
            ),
          })
        }
      />
    </View>
  );
};

const styleFn = createStyles(({spaces, colors}) => ({
  container: {gap: spaces.Space8},
  ref: {
    alignItems: 'center',
    padding: spaces.Space8,
    backgroundColor: colors.BgAccent,
  },
}));
```

## `Props`

### <div class="label required basic">Required</div>**`title`**

| TYPE   |
|:-------|
| string |

Title for tooltip

### `titleStyles`

| TYPE                                                       |
|:-----------------------------------------------------------|
| [TextStyle](https://reactnative.dev/docs/text-style-props) |

Custom styles for container

### `titleFont`

| Type                                                                                                       | DEFAULT                |
|:-----------------------------------------------------------------------------------------------------------|:-----------------------|
| [TypographyProp](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/typography/types.ts#L49) | 'SemiBold-Secondary-M' |

Font for title
