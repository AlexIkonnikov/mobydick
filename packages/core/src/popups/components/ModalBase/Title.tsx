import React, {FC} from 'react';

import {ITitlePopup} from '../PopupBase';
import useStyles from '../../../styles/theme/hooks/useStyles';
import {Typography} from '../../../typography/components/Typography/Typography';

import stylesCreate from './stylesCreate';

/**
 * @deprecated use TextContent.tsx
 */

const Title: FC<ITitlePopup> = props => {
  const {title, titleStyles, titleFont} = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <Typography
      style={[styles.text, titleStyles]}
      font={titleFont ? titleFont : 'SemiBold-Primary-L'}>
      {title}
    </Typography>
  );
};

export default Title;
