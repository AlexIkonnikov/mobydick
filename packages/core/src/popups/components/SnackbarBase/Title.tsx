import {FC} from 'react';

import useStyles from '../../../styles/hooks/useStyles';
import {Typography} from '../../../typography/components/Typography/Typography';
import {ITitlePopup} from '../PopupBase';

import stylesCreate from './stylesCreate';

const Title: FC<ITitlePopup> = props => {
  const [styles] = useStyles(stylesCreate);
  const {title, titleStyles, titleFont} = props;

  return (
    <Typography
      style={[styles.title, titleStyles]}
      font={titleFont ? titleFont : 'Medium-Contrast-XS'}>
      {title}
    </Typography>
  );
};

export default Title;
