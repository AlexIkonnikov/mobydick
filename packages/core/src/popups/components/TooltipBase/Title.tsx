import {FC} from 'react';

import {ITitlePopup} from '../PopupBase';
import useStyles from '../../../styles/hooks/useStyles';
import {Typography} from '../../../typography/components/Typography/Typography';

import stylesCreate from './stylesCreate';

const Title: FC<ITitlePopup> = props => {
  const [styles] = useStyles(stylesCreate);
  const {title, titleStyles, titleFont} = props;

  return (
    <Typography
      font={titleFont ? titleFont : 'SemiBold-Contrast-XS'}
      style={[styles.title, titleStyles]}>
      {title}
    </Typography>
  );
};

export default Title;
