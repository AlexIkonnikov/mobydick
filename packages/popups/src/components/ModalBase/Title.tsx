import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';
import {Typography} from '@npm/mobydick-typography';

import {ITitlePopup} from '../PopupBase';

import stylesCreate from './stylesCreate';

/**
 * @deprecated use TextContent.tsx
 */

const Title: FC<ITitlePopup> = props => {
  const {title, titleStyles, titleFont} = props;
  const [styles] = useStyles(stylesCreate);

  return (
    <Typography
      style={[styles.text, titleStyles]}
      font={titleFont ? titleFont : 'SemiBold-Primary-L'}>
      {title}
    </Typography>
  );
};

export default Title;
