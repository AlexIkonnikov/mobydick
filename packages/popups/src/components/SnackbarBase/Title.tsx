import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';

import {ITitlePopup} from '../PopupBase';
import {Typography} from '../../../../typography';

import stylesCreate from './stylesCreate';

const Title: FC<ITitlePopup> = props => {
  const [styles] = useStyles(stylesCreate);
  const {title, titleStyles} = props;

  return <Typography style={[styles.title, titleStyles]}>{title}</Typography>;
};

export default Title;
