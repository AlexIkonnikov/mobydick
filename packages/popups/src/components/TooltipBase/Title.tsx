import React, {FC} from 'react';
import {useStyles} from '@npm/mobydick-styles';

import {ITitlePopup} from '../PopupBase';
import {Typography} from '../../../../typography';

import stylesCreate from './stylesCreate';

const Title: FC<ITitlePopup> = props => {
  const [styles] = useStyles(stylesCreate);
  const {title, titleStyles, titleFont} = props;

  return (
    <Typography
      font={titleFont ? titleFont : 'Regular-Primary-XS'}
      style={[styles.title, titleStyles]}>
      {title}
    </Typography>
  );
};

export default Title;
