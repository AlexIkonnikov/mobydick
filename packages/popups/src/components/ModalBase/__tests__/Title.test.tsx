import React from 'react';
import {render} from '@testing-library/react-native';

import Title from '../Title';

describe('@npm/mobydick-popups/Title', () => {
  afterEach(() => {
    jest.resetAllMocks();
    jest.clearAllMocks();
  });
  it('should renders correctly', () => {
    const {toJSON} = render(<Title title={'title'} titleStyles={{flex: 1}} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
