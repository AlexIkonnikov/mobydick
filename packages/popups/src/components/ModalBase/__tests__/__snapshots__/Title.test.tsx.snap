// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/Title should renders correctly 1`] = `
<Text
  allowFontScaling={true}
  style={
    Array [
      Object {
        "color": "#20242D",
        "fontSize": 36,
        "fontWeight": "600",
        "lineHeight": 39,
        "paddingHorizontal": 30,
        "paddingVertical": 15,
        "textAlign": "center",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  title
</Text>
`;
